test/jdk/javax/swing/JInternalFrame/InternalFrameBorderTest.java

```diff
@@ -0,0 +1,248 @@
+ */
+
+/*
+ * @test
+ * @bug 8015739
+ * @key headful
+ * @summary Tests whether background color of JInternalFrame is visible
+ * in the border region at different scales by checking the midpoints
+ * and corners of the border.
+ *
+ * @requires (os.family == "mac" | os.family == "linux")
+ * @run main/othervm -Dsun.java2d.uiScale=1 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=2 InternalFrameBorderTest
+ */
+
+public class InternalFrameBorderTest {
+    private static final int FRAME_SIZE = 300;
+    private static final int INTFRAME_SIZE = 200;
+    private static final int MIDPOINT = INTFRAME_SIZE/2;
+    private static final int BORDER_THICKNESS = 5;
+    private static StringBuffer errorLog = new StringBuffer();
+
+    private static JFrame jFrame;
+    private static JInternalFrame iFrame;
+    private static Robot robot;
+
+    private static Point iFrameLoc;
+    private static int iFrameMaxX;
+    private static int iFrameMaxY;
+    private static String uiScale;
+
+    public static void main(String[] args) throws AWTException,
+            InterruptedException, InvocationTargetException {
+        try {
+            UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        } catch (Exception e) {
+            System.out.println("Metal LAF class not supported");
+            return;
+        }
+
+        try {
+            robot = new Robot();
+            robot.setAutoDelay(200);
+            uiScale = System.getProperty("sun.java2d.uiScale");
+
+            SwingUtilities.invokeAndWait(InternalFrameBorderTest::createAndShowGUI);
+            robot.waitForIdle();
+            robot.delay(500);
+
+            SwingUtilities.invokeAndWait(() -> {
+                iFrameLoc = iFrame.getLocationOnScreen();
+                iFrameMaxX = iFrameLoc.x + INTFRAME_SIZE;
+                iFrameMaxY = iFrameLoc.y + INTFRAME_SIZE;
+            });
+
+            // Check Borders
+            SwingUtilities.invokeAndWait(() -> {
+                checkBorderMidPoints("TOP");
+                checkBorderMidPoints("RIGHT");
+                checkBorderMidPoints("BOTTOM");
+                checkBorderMidPoints("LEFT");
+            });
+
+            // Check Corners
+            SwingUtilities.invokeAndWait(() -> {
+                checkCorners("TOP_LEFT");
+                checkCorners("TOP_RIGHT");
+                checkCorners("BOTTOM_RIGHT");
+                checkCorners("BOTTOM_LEFT");
+            });
+            if (!errorLog.isEmpty()) {
+                throw new RuntimeException("Following error(s) occurred: \n"
+                        + errorLog);
+            }
+        } finally {
+            if (jFrame != null) {
+                jFrame.dispose();
+            }
+            robot.delay(500);
+        }
+    }
+
+    private static void checkBorderMidPoints(String borderDirection) {
+        int x = 0, y = 0, start = 0, stop = 0;
```

Comment:
The initialisers are redundant.  
It's not recommended to have several local variable declarations on one line.

When using two lines
```java
int x, y;
int start, stop;
```
the coordinates and loop control variables are on separate lines, easier to parse.